src/cli/find-cache-file.js

```diff
@@ -3,17 +3,40 @@
 const os = require("os");
 const path = require("path");
 const findCacheDir = require("find-cache-dir");
+const { statSafe } = require("./utils.js");
 
 /**
  * Find default cache file (`./node_modules/.cache/prettier/.prettier-cache`) using https://github.com/avajs/find-cache-dir
- *
- * @returns {string}
  */
-function findCacheFile() {
+function findDefaultCacheFile() {
   const cacheDir =
     findCacheDir({ name: "prettier", create: true }) || os.tmpdir();
   const cacheFilePath = path.join(cacheDir, ".prettier-cache");
   return cacheFilePath;
 }
 
+async function findCacheFileFromOption(cacheLocation) {
+  const cwd = process.cwd();
+  const normalized = path.normalize(cacheLocation);
+  const cacheFile = path.join(cwd, normalized);
```

Comment:
Should we run `path.isAbsolute()` check first? Not sure how it works for `/foo` and `D:\foo`(Windows, different partition))

Reply:
`process.cwd()` returns absolute path. So I don't think we should run `path.isAbsolute` check.

Reply:
I didn't mean check cwd, `cacheLocation`.

Reply:
Maybe  `path.resolve(cacheLocation)` is enough.

Reply:
[c5b2063](https://github.com/prettier/prettier/pull/13019/commits/c5b20630b8c6b71eb91733959f9aea185c6bd390)